File: src/main/java/com/uiptv/model/Bookmark.java
package com.uiptv.model;

import com.uiptv.shared.BaseJson;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class Bookmark extends BaseJson {
    private String dbId;
    private String accountName;
    private String categoryTitle;
    private String channelId;
    private String channelName;
    private String cmd;
    private String serverPortalUrl;
    private String categoryId;
    private Account.AccountAction accountAction; // Added field
    private String drmType;
    private String drmLicenseUrl;
    private String clearKeysJson;
    private String inputstreamaddon;
    private String manifestType;

    public Bookmark(String accountName, String categoryTitle, String channelId, String channelName, String cmd, String serverPortalUrl, String categoryId) {
        this.accountName = accountName;
        this.categoryTitle = categoryTitle;
        this.channelId = channelId;
        this.channelName = channelName;
        this.cmd = cmd;
        this.serverPortalUrl = serverPortalUrl;
        this.categoryId = categoryId;
    }

    public void setFromChannel(Channel channel) {
        this.setDrmType(channel.getDrmType());
        this.setDrmLicenseUrl(channel.getDrmLicenseUrl());
        this.setClearKeysJson(channel.getClearKeysJson());
        this.setManifestType(channel.getManifestType());
        this.setInputstreamaddon(channel.getInputstreamaddon());
    }
}

File: server/src/main/kotlin/com/uiptv/model/Bookmark.kt
package com.uiptv.model

import com.uiptv.shared.BaseJson

class Bookmark() : BaseJson() {
    var dbId: String? = null
    var accountName: String? = null
    var categoryTitle: String? = null
    var channelId: String? = null
    var channelName: String? = null
    var logo: String? = null
    var cmd: String? = null
    var serverPortalUrl: String? = null
    var categoryId: String? = null
    var accountAction: Account.AccountAction? = null
    var drmType: String? = null
    var drmLicenseUrl: String? = null
    var clearKeysJson: String? = null
    var inputstreamaddon: String? = null
    var manifestType: String? = null
    var categoryJson: String? = null
    var channelJson: String? = null
    var vodJson: String? = null
    var seriesJson: String? = null

    constructor(
        accountName: String?,
        categoryTitle: String?,
        channelId: String?,
        channelName: String?,
        cmd: String?,
        serverPortalUrl: String?,
        categoryId: String?
    ) : this() {
        this.accountName = accountName
        this.categoryTitle = categoryTitle
        this.channelId = channelId
        this.channelName = channelName
        this.cmd = cmd
        this.serverPortalUrl = serverPortalUrl
        this.categoryId = categoryId
    }

    fun setFromChannel(channel: Channel) {
        logo = channel.logo
        drmType = channel.drmType
        drmLicenseUrl = channel.drmLicenseUrl
        clearKeysJson = channel.clearKeysJson
        manifestType = channel.manifestType
        inputstreamaddon = channel.inputstreamaddon
    }
}
